miniwin/miniwin/src/miniwin_d3drmmesh.cpp

```diff
@@ -1,19 +1,26 @@
 #include "miniwin_d3drmmesh_p.h"
 #include "miniwin_p.h"
 
+#include <algorithm>
+
 HRESULT Direct3DRMMeshImpl::Clone(int flags, GUID iid, void** object)
 {
-	if (SDL_memcmp(&iid, &IID_IDirect3DRMMesh, sizeof(GUID)) == 0) {
-		*object = static_cast<IDirect3DRMMesh*>(new Direct3DRMMeshImpl);
-		return DD_OK;
+	if (!object || SDL_memcmp(&iid, &IID_IDirect3DRMMesh, sizeof(GUID)) != 0) {
+		return DDERR_INVALIDPARAMS;
 	}
 
-	return DDERR_GENERIC;
-}
+	auto* clone = new Direct3DRMMeshImpl(*this);
 
-HRESULT Direct3DRMMeshImpl::GetBox(D3DRMBOX* box)
-{
-	MINIWIN_NOT_IMPLEMENTED();
+	for (auto& [_, group] : clone->m_groups) {
+		if (group.texture) {
+			group.texture->AddRef();
+		}
+		if (group.material) {
+			group.material->AddRef();
+		}
```

Comment:
I'm not sure this is the correct approach.
Perhaps the material should also be duplicated.

Perhaps add a comment documenting this concern?

Reply:
I was unable to find docs to say one way or the other, this seemed the most reasonable.